libbeat/autodiscover/template/config.go

```diff
@@ -43,6 +45,10 @@ type MapperSettings []*struct {
 
 // NewConfigMapper builds a template Mapper from given settings
 func NewConfigMapper(configs MapperSettings) (mapper Mapper, err error) {
+	if len(configs) == 0 {
+		return nil, fmt.Errorf("no configs defined for autodiscover provider")
+	}
+
 	for _, c := range configs {
 		condMap := &ConditionMap{Configs: c.Configs}
```

Comment:
Another check we could do here is to ensure `len(c.Configs) > 0`. I think in most cases this is were bad configs happen? when the user does:

```
autodiscover:
  providers:
    type: kubernetes
    templates:
      - equals:
          container.image: ...
        config:
          type: container
```

But config is expected to be a list, so a `-` is missing

Reply:
I'm a bit confused, is this section used when hints are enabled?